maintenance/importConfig.ts

```diff
@@ -835,6 +975,45 @@ ${JSON.stringify(parsed, undefined, 4)}`;
 	}
 }
 
+/** Get the latest version from file name */
+function parseFileName(fileName: string): SemVer | string | null {
+	fileName = path.basename(fileName, ".json");
+
+	const versionRange = fileName.split("_")[1]?.split("-");
+
+	const min = versionRange ? coerce(versionRange[0]) : "0.0.0";
+	const max = versionRange ? coerce(versionRange[1]) : null;
+
+	return max ?? min;
+}
+
+/**
+ * Get latest device configuration file version
+ * @param manufacturerDir manufacturer folder
+ * @param productName  product name
+ */
+async function getLatestConfigVersion(
+	manufacturerDir: string,
+	productName: string,
+): Promise<string | undefined> {
+	// get all files in manufacturer dir related to this product
+	// could be none, one, or multiple files if there are many versions
+	const files = (await fs.readdir(manufacturerDir)).filter(
+		(f) => f === `${productName}.json` || f.startsWith(productName + "_"),
+	);
+
+	// sort the files by parsing the version range
+	// <productname>_<minVersion?>-<maxVersion>.json
+	files.sort((a, b) => {
+		const vA = parseFileName(a) ?? "0.0.0";
+		const vB = parseFileName(b) ?? "0.0.0";
+
+		return compare(vA, vB);
+	});
+
+	return files[files.length - 1];
+}
+
```

Comment:
I think its better to use the index to look this up instead of parsing file names, similar to:
https://github.com/zwave-js/node-zwave-js/blob/5d5677a0d92ece7db18014cf86e8fafd64d89bb5/packages/config/src/Devices.ts#L94-L115
Then take the highest version from the index entries

Reply:
@AlCalzone I don't understand how to use that to achieve the same result, when I call that method I need to specify also the firmware version

Reply:
Sorry, was a bit unclear. I'm suggesting to use
```ts
index.filter(entry => entry.manufacturerId === product.manufacturerId && entry.productType === product.productType && entry.productId === product.productId)
```
to find all matching config files and from that list select the one with the highest firmware version.

The filename-based lookup will break when we decide to rename the files or folders at some point.

Reply:
I have started using it, but seems there are files with same man, prod type and id but different names :confused: